api/data_test.go

```diff
@@ -149,3 +149,88 @@ func TestCreateNewCacheTime(t *testing.T) {
 		})
 	})
 }
+
+func TestCreateOrUpdateCacheTimeReturnsErr(t *testing.T) {
+	const validBody = `{"path": "testpath", "etag": "testetag"}`
+	Convey("When no request body is provided, 400 bad request is returned", t, func() {
```

Comment:
All the other tests have been written in a Given/When/Then fashion, so it may be worth refactoring these new ones to match ONS' preferred style.

Reply:
Myself and Marilia tried to follow the convention used in [dp-dataset-api](https://github.com/ONSdigital/dp-dataset-api/blob/develop/api/dataset_test.go#L456). Could you elaborate a little bit on the implementation of this please?

Reply:
Sure. If you check the rest of the tests of this file, you'll see that they all tend to have 3 `Convey` blocks: the first one is the "Given", which sets the test up. Then there is the "When" block, which runs the code. Finally, there is the "Then" block, which runs the assertions, checking that the results are what you expected.

If you search for Given Then When, you'll find a lot of resources, as it's a well known pattern. It's very similar to the AAA pattern (Arrange, Act, Assert).

Reply:
Have updated the tests - let me know if this is more in line with what you mean.